giskard/models/langchain.py

```diff
@@ -38,15 +38,32 @@ def __init__(
             **kwargs,
         )
 
-    def save_model(self, local_path, mlflow_meta):
-        mlflow.langchain.save_model(self.model, path=local_path, mlflow_model=mlflow_meta)
+    def save(self, local_path: Union[str, Path]) -> None:
+        super().save(local_path)
+        self.save_model(local_path)
+
+    def save_model(self, local_path: Union[str, Path]) -> None:
+        path = Path(local_path)
+        self.model.save(path / "chain.json")
 
     @classmethod
-    def load_model(cls, local_dir):
-        return mlflow.langchain.load_model(local_dir)
+    def load_model(cls, local_dir, **kwargs):
+        from langchain.chains import load_chain
+
+        path = Path(local_dir)
+        return load_chain(path / "chain.json", **kwargs)
 
     def model_predict(self, df):
-        return [self.model.predict(**data) for data in df.to_dict("records")]
+        generations = [self.model(data) for data in df.to_dict("records")]
```

Comment:
It's actually a bug fix. The former method only works with `LLMChain` while the new one works with any type extending `Chain`, this will continue to work with models uploaded with previous version

Reply:
Is it related to this as well you think? https://linear.app/giskard/issue/GSK-1819/set-a-minimum-on-langchains-version